Fix backwards compatibility issues with dbus's plist files

zbentley opened the pull request:
### Description

D-Bus includes a session daemon plist file. However, homebrew is not
explicitly told about this file (though it figures it out),and the file
contains the ServiceIPC key, which will emit "deprecated" warnings
in launchd on modern OSX. This commit patches the included config file
based on OSX version to ensure that files free of legacy cruft are
distributed correctly.


Other changes included:
- Removed pointless "chmod" of the distributed plist file. Unless built as root, the generated plist has appropriate permissions in standard Homebrew installs. 
- Added "master" branch info for the recommended public-pull source repo for D-Bus.

### All Submissions:

- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew/pulls) for the same update/change?

_You can erase any parts of this template not applicable to your Pull Request._

### New Formulae Submissions:

- [X] Does your submission pass
`brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [X] Have you built your formula locally prior to submission with `brew install <formula>`?

dunn commented on Library/Formula/d-bus.rb:
```diff
@@ -4,6 +4,7 @@ class DBus < Formula
   homepage "https://wiki.freedesktop.org/www/Software/dbus"
   url "https://dbus.freedesktop.org/releases/dbus/dbus-1.10.6.tar.gz"
   mirror "https://mirrors.ocf.berkeley.edu/debian/pool/main/d/dbus/dbus_1.10.6.orig.tar.gz"
+  master "git://anongit.freedesktop.org/dbus/dbus.git"
```
We use `head`, not `master`.

dunn commented on Library/Formula/d-bus.rb:
```diff
@@ -41,4 +57,4 @@ def post_install
     system "#{bin}/dbus-daemon", "--version"
   end
-end
+end
```
Could you re-add the trailing newline?

zbentley commented:
FYI, unless I'm reading the `brew audit` output wrong, it's warning about the trailing newline; that's why I deleted it.

DomT4 commented:
The core doesn't even support Leopard and below these days. Let's not bother patching that far back.

DomT4 commented on Library/Formula/d-bus.rb:
```diff
@@ -12,6 +13,17 @@ class DBus < Formula
+    sha256 "a8aa6fe3f2d8f873ad3f683013491f5362d551bf5d4c3b469f1efbc5459a20dc"
+  end
+
+  def plist_name
```
Where/why is this variable being used?

DomT4 commented on Library/Formula/d-bus.rb:
```diff
@@ -12,6 +13,17 @@ class DBus < Formula
+  # Patch applies the config templating fixed in https://bugs.freedesktop.org/show_bug.cgi?id=94494
+  # Homebrew pr/issue: 50219
+  patch do
+    url "https://raw.githubusercontent.com/zbentley/dbus-osx-examples/master/homebrew-patches/org.freedesktop.dbus-session.plist.osx.diff"
```
Please submit this to our patches repo over at https://github.com/Homebrew/patches and use that link instead, Thanks!

DomT4 commented on Library/Formula/d-bus.rb:
```diff
@@ -29,8 +41,6 @@ def install
     ENV.deparallelize
     system "make", "install"
-
-    (prefix/"org.freedesktop.dbus-session.plist").chmod 0644
```
No longer needs the `chmod`?

dunn replied:
This question remains.

zbentley replied:
Unless users have nonstandard permissions on their homebrew installation directories this is not necessary. The D-Bus make/config scripts will not chmod that file; it appropriately inherits permissions when installed without the chmod.

DomT4 commented on Library/Formula/d-bus.rb:
```diff
@@ -5,6 +5,7 @@ class DBus < Formula
   url "https://dbus.freedesktop.org/releases/dbus/dbus-1.10.6.tar.gz"
   mirror "https://mirrors.ocf.berkeley.edu/debian/pool/main/d/dbus/dbus_1.10.6.orig.tar.gz"
   sha256 "b5fefa08a77edd76cd64d872db949eebc02cf6f3f8be82e4bbc641742af5d35f"
+  head "git://anongit.freedesktop.org/dbus/dbus.git"
```
Should be able to use `https://` on this:

```ruby
head "https://anongit.freedesktop.org/git/dbus/dbus.git"
```
?

zbentley commented:
> No longer needs the `chmod`?

Nope. Unless someone is running nonstandard permissions on their homebrew directory.

> Should be able to use `https://` on this:

Doesn't seem to work with anything but `git:`:
```bash
zbentley@zbentley-mbp13 ~ ∴ git clone https://anongit.freedesktop.org/dbus/dbus.git
Cloning into 'dbus'...
fatal: repository 'https://anongit.freedesktop.org/dbus/dbus.git/' not found
zbentley@zbentley-mbp13 ~ ∴ git clone https://anongit.freedesktop.org/dbus/dbus
Cloning into 'dbus'...
fatal: repository 'https://anongit.freedesktop.org/dbus/dbus/' not found
zbentley@zbentley-mbp13 ~ ∴ git clone git://anongit.freedesktop.org/dbus/dbus.git
Cloning into 'dbus'...
remote: Counting objects: 30220, done.
...
```

> Please submit this to our patches repo over at https://github.com/Homebrew/patches and use that link instead, Thanks!

Done; pending approval in homebrew/patches#15. I submitted the patch for the old OSX there as well in hopes that the back-compat homebrew stuff in  mistydemeo/tigerbrew can reference it. If that's inappropriate I'll revise the patches PR.

Submitting changes to this PR to use the patches momentarily; it won't be testable until the patches/#15 gets merged though.

> Where/why is this variable being used?

It's not, I think? I was under the impression that there was some part of homebrew that was aware of plist files declared in this fashion and kept track of them. Apparently not; I'll remove it.

dunn commented:
The URL for `https` is a little different, but it works: https://anongit.freedesktop.org/git/dbus/dbus.git

DomT4 commented on Library/Formula/d-bus.rb:
```diff
@@ -12,6 +13,13 @@ class DBus < Formula
+  # Patch applies the config templating fixed in https://bugs.freedesktop.org/show_bug.cgi?id=94494
+  # Homebrew pr/issue: 50219
+  patch do
+    url "https://raw.githubusercontent.com/homebrew/patches/master/d-bus/org.freedesktop.dbus-session.plist.osx.diff"
```
Please use the commit reference rather than `master`. Means if it ever gets accidentally removed this formula doesn't start randomly dying on `fetch` for people.

zbentley commented:
> The URL for https is a little different, but it works: https://anongit.freedesktop.org/git/dbus/dbus.git

Fixed; sorry for missing that.

> Please use the commit reference rather than master. Means if it ever gets accidentally removed this formula doesn't start randomly dying on fetch for people.

Fixed to reference file by the first chunk of the commit hash.

Both fixes are in 78f1bff.

DomT4 commented:
Merged in https://github.com/Homebrew/homebrew/commit/06f9b0376fb997a2bb30ecb02dd68ad903e1b00c. Thank you for your contribution to Homebrew @zbentley; we appreciate it! :smiley_cat:
